### PR TITLE
Added priority handling. Closes #81

### LINKED ISSUE #81: Allow priority specification for @GuideBook
I have a mod and an addon for it that both use guide books with `@GuideBook`. I need the addon's guide book to load after the base mod's, and right now, that isn't happening. I see two ways to solve this. The first one, which would probably take a lot more work, is to figure out which mods add which books, and load them in the same order Forge loads the mods.
The second, and likely easier option, would me to make it so modders using `@GuideBook` can specify a priority in parenthesis, and if they don't, it is normal by default. Much like how `@SubscribeEvent` does things.

### PR DESCRIPTION
**Pull request by The-Fireplace**
I was unable to test this change due to [this crash](http://paste.ubuntu.com/25186191/) when I attempted to run the client. It appears to be related to #77, I can confirm that it was not caused by these changes.
EDIT: I am going to disable the part causing the crash for testing purposes only. I will comment once I have tested my changes.
EDIT 2: Tested and confirmed working!

### CODE REVIEW

**Comment by The-Fireplace** [comment]
Tested and confirmed working!

**Comment by TehNut** [comment]
Oh, I have that fixed locally

File: src/main/java/amerifrance/guideapi/util/AnnotationHandler.java [hunk 1]
```diff
@@ -6,32 +6,46 @@
...
-                if (book == null)
-                    continue;
-                APISetter.registerBook(book);
-                BOOK_CLASSES.add(Pair.of(book, guideBook));
-            } catch (Exception e) {
-                LogHelper.error("Error registering book for class " + data.getClassName());
-                e.printStackTrace();
+                    IGuideBook guideBook = (IGuideBook) genericClass.newInstance();
+                    boolean wrongPriority = false;
+                    for (Annotation f: genericClass.getAnnotations()) {
```
Comment (TehNut) [inline src/main/java/amerifrance/guideapi/util/AnnotationHandler.java]:
```java
ModAnnotation.EnumHolder holder = (ModAnnotation.EnumHolder) data.getAnnotationInfo().get("priority");
EventPriority bookPriority = holder == null ? EventPriority.NORMAL : EventPriority.valueOf(holder.getValue());
boolean wrongPriority = priority != bookPriority;
```

**Review by TehNut — CHANGES_REQUESTED** [review]
Rather than looking through the class annotations, just pull from the data that Forge provides.

File: src/main/java/amerifrance/guideapi/util/AnnotationHandler.java (hunk 1 above)
Comment (The-Fireplace) [reply src/main/java/amerifrance/guideapi/util/AnnotationHandler.java]: Done

**Review by TehNut — APPROVED** [review]

**Pull request merged**